### PR TITLE
cleanup code and refactor

### PR DESCRIPTION
**Pull request by ylwu-amzn**
Signed-off-by: Yaliang Wu <ylwu@amazon.com>

### Description
Clean up unused/unnecessary code and refactor.
 
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: common/src/main/java/org/opensearch/ml/common/parameter/MLPredictionOutput.java [hunk 1]
```diff
@@ -82,7 +82,7 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         if (taskId != null) {
             builder.field(TASK_ID_FIELD, taskId);
         }
-        if (status != null){
+        if (status != null) {
```
Comment (wnbts) [inline common/src/main/java/org/opensearch/ml/common/parameter/MLPredictionOutput.java:85]: btw: we should have checkstyle, lincense checks, etc for this repo.

File: plugin/src/main/java/org/opensearch/ml/task/MLTaskRunner.java [hunk 2]
```diff
@@ -14,23 +14,25 @@
...
 import org.opensearch.ml.model.MLTask;
 import org.opensearch.ml.model.MLTaskState;
 import org.opensearch.ml.stats.MLStats;
+import org.opensearch.transport.TransportService;
 
 /**
  * MLTaskRunner has common code for dispatching and running predict/training tasks.
  */
-public class MLTaskRunner {
+public abstract class MLTaskRunner<S, T> {
```
Comment (wnbts) [inline plugin/src/main/java/org/opensearch/ml/task/MLTaskRunner.java:26]: minor. generic type parameters can also be part of javadoc.

**Review by wnbts — APPROVED** [review]
looks good to me. thanks for the quick changes!

**Review by jackiehanyang — APPROVED** [review]

File: common/src/main/java/org/opensearch/ml/common/parameter/MLPredictionOutput.java (hunk 1 above)
Comment (ylwu-amzn) [reply common/src/main/java/org/opensearch/ml/common/parameter/MLPredictionOutput.java:85]: yes, totally agree, create an issue to improve checkstyle

File: plugin/src/main/java/org/opensearch/ml/task/MLTaskRunner.java (hunk 2 above)
Comment (ylwu-amzn) [reply plugin/src/main/java/org/opensearch/ml/task/MLTaskRunner.java:26]: thanks, missed this comment before merging, will fix in next PR

**Pull request merged**